.github/workflows/release.yml

```
@@ -0,0 +1,152 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build And Publish Release
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+jobs:
+  push:
+    permissions:
+      contents: read
+      id-token: write # this is important, it's how we authenticate with Vault
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - linux
+        arch:
+          - amd64
+          - arm64
+    steps:
+      - name: Prepare Matrix Instance
+        run: |
+          echo "PLATFORM_PAIR=${{ matrix.os }}-${{ matrix.arch }}"
+          echo "PLATFORM_PAIR=${{ matrix.os }}-${{ matrix.arch }}" >> $GITHUB_ENV
+          echo "GOOS=${{ matrix.os }}"
+          echo "GOOS=${{ matrix.os }}"     >> $GITHUB_ENV
+          echo "GOARCH=${{ matrix.arch }}"
+          echo "GOARCH=${{ matrix.arch }}" >> $GITHUB_ENV
+          echo "ARCH=${{ matrix.arch }}"
+          echo "ARCH=${{ matrix.arch }}"   >> $GITHUB_ENV
+
+      - name: Retrieve Docker Hub Credentials From Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKERHUB_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKERHUB_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ env.DOCKERHUB_PASSWORD }}
+
+      - name: Get Sources
+        uses: actions/checkout@v4
+
+      - name: Docker Meta Data Setup
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          cache: false
```

We should use the go.mod version with like `ci.yml`
```suggestion
      - name: Set up Go
        uses: actions/setup-go@v5
        with:
          go-version-file: 'go.mod'
          cache: false
```

---

Will fix.

---

Fixed with commit dd3c282ee030db5090de1eeda91109eba4f1786a